Adding Cerberus integration

dry923 opened the pull request:
This adds an optional CERBERUS_URL variable that can be passed to all install/upgrade/scale jobs. If enabled then it will check the status of the cluster prior and post activity. It fails if Cerberus returns False during any of the checks

dry923 commented:
Do we want to enable this for OCP 3.x as well?

chaitanyaenr commented on OCP-4.X/install-on-aws.yml:
```diff
@@ -20,6 +20,8 @@
     - role: post-config-on-aws
       when: openshift_post_config|bool
+    - role: cerberus
+      when: cerberus_url is defined and cerberus_url != ""
```
Maybe also check if cerberus is enabled - https://github.com/openshift-scale/scale-ci-deploy/blob/master/OCP-4.X/vars/install-on-aws.yml#L98?

chaitanyaenr commented on OCP-4.X/vars/install-on-aws.yml:
```diff
@@ -101,3 +101,6 @@ openshift_alertmanager_storage_size: "{{ lookup('env', 'OPENSHIFT_ALERTMANAGER_S
 ## use only when NOT using the default auth dir located in "${GOPATH}/src/github.com/openshift/installer"
 kubeconfig_auth_dir_path: "{{ lookup('env', 'KUBECONFIG_AUTH_DIR_PATH') }}"
+
+## To enable Cerberus integration add the url of cerberus in the format "http://1.2.3.4:8080"
```
We might want to add a note that the cerberus_url can be defined and used to check on things even when the Cerberus is installed outside scale-ci-deploy ( https://github.com/openshift-scale/scale-ci-deploy/blob/master/OCP-4.X/vars/install-on-aws.yml#L98 ).

chaitanyaenr commented on OCP-4.X/vars/install-on-aws.yml:
```diff
@@ -101,3 +101,6 @@ openshift_alertmanager_storage_size: "{{ lookup('env', 'OPENSHIFT_ALERTMANAGER_S
 kubeconfig_auth_dir_path: "{{ lookup('env', 'KUBECONFIG_AUTH_DIR_PATH') }}"
+
+## To enable Cerberus integration add the url of cerberus in the format "http://1.2.3.4:8080"
+cerberus_url: "{{ lookup('env', 'CERBERUS_URL')|default('') }}"
```
Maybe move the var under other cerberus vars which got added recently?

dry923 replied:
agreed. I'm also adding a section to define it to the localhost ip if it's created in the post install config. Updated PR should be in on Monday.

chaitanyaenr replied:
It runs the http server and exposes the signal at http://0.0.0.0:8080, this way we can access it using the same address assuming we are hitting the url on the same host as cerberus. In case the cerberus host and host from where we are pinging it are different, then we can use the public ip of the cerberus host to get the signal.

dry923 commented:
I updated the PR to account for the recent cerberus install additions. I also, moved the cerberus install section of the post install to its own role to allow for easier potential reuse later.
